### DISCUSSION TITLE
How to get response headers size in bytes

### DISCUSSION THREAD

**Discussion by let4be** [hyperium/hyper#2502] [Q&A]
For statistical purposes I would like to record exact size of headers in an http response(n of bytes), is this possible?

I guess I could estimate this size by running across `headers()` and summing string lengths, but this seems dirty and slow. Is there a better way?

**Comment by seanmonstar** [comment]
The number would be different depending on HTTP/1 and HTTP/2. In HTTP/1, hyper emits a trace event about bytes read and parsed into headers. We could probably make that a slightly better structured `tracing` event with better key/values, so that you can then grab it in a tracing subscriber.

**Comment by let4be** [comment]
Thinking more about it I'm trying to make a wrapper around connector(`hyper_tls::HttpsConnector`) and simply record n of bytes transferred and received this should be exactly what I need for my statistical purposes - I will get an exact total size of request/response this way(separating headers, body not really necessary, more interested in the total transfered/received over a socket for particular request/response)

Now even if this was in a tracing event, I do not understand how I could catch this information with the desired context as I need this to be recorded in specific place of my logic, enriched(with context from my other data) and further emitted to a separate thread to be stored as metrics...
